src/zope/untrustedpython/tests.py

```diff
@@ -112,69 +124,60 @@ def test_CompiledProgram_simple(self):
+        exec(rcompile.compile("x = 1", "<string>", "exec"), globals())
+        # `exec` put `x` into the globals:
         self.assertEqual(x, 1)  # noqa: F821 undefined name 'x'
 
     def test_CompiledCode_no_exec(self):
-        self.assertRaises(
-            SyntaxError, rcompile.compile,
-            "exec 'x = 2'", "<string>", "exec")
-        self.assertRaises(
-            SyntaxError, rcompile.compile,
-            "raise KeyError('x')", "<string>", "exec")
-        self.assertRaises(
-            SyntaxError, rcompile.compile,
-            "try: pass\nexcept: pass", "<string>", "exec")
-
-    def test_CompiledCode_explicit_writable(self):
-        import StringIO
-        f = StringIO.StringIO()
-        code = rcompile.compile(
-            "print >> f, 'hi',\nprint >> f, 'world'", '', 'exec')
-        exec code in {'f': f}
-        self.assertEqual(f.getvalue(), 'hi world\n')
-
-    def test_CompiledCode_default_output(self):
-        def _exec(code, locals):
-            exec code in locals
-        code = rcompile.compile("print 'hi',\nprint 'world'", '', 'exec')
-        self.assertRaises(NameError, _exec, code, {})
-        import StringIO
-        f = StringIO.StringIO()
-        _exec(code, {'untrusted_output': f})
-        self.assertEqual(f.getvalue(), 'hi world\n')
-
-
-class Test_RestrictionMutator(unittest.TestCase):
-
-    def test_error_no_line_info(self):
-        rm = rcompile.RestrictionMutator()
-        rm.error(None, 'nothing')
-        self.assertEqual(rm.errors, ['nothing'])
+        with self.assertRaises(SyntaxError) as err:
+            rcompile.compile("exec('x = 2')", "<string>", "exec")
+        if six.PY2:
+            self.assertEqual("('Line 1: Exec statements are not allowed.',)",
+                             str(err.exception))  # pragma: PY2
+        else:
+            self.assertEqual("('Line 1: Exec calls are not allowed.',)",
+                             str(err.exception))  # pragma: PY3
+        with self.assertRaises(SyntaxError) as err:
+            rcompile.compile("raise KeyError('x')", "<string>", "exec")
+        self.assertEqual("('Line 1: Raise statements are not allowed.',)",
+                         str(err.exception))
+        with self.assertRaises(SyntaxError) as err:
+            rcompile.compile("try: pass\nexcept: pass", "<string>", "exec")
+        if six.PY2:
+            self.assertEqual(
+                "('Line 1: TryExcept statements are not allowed.',)",
+                str(err.exception))  # pragma: PY2
+        else:
+            self.assertEqual(
+                "('Line 1: Try statements are not allowed.',)",
+                str(err.exception))  # pragma: PY3
 
 
 def test_suite():
     return unittest.TestSuite((
         unittest.makeSuite(Test_SafeBuiltins),
         unittest.makeSuite(Test_Interpreter),
         unittest.makeSuite(Test_Compiled),
-        unittest.makeSuite(Test_RestrictionMutator),
     ))
```

Comment:
How about

```python
def test_suite():
    return unittest.defaultTestLoader.loadTestsFromName(__name__)
```
?

Or, actually, that's the default for modern zope.testrunner when no custom test_suite is defined, so we might as well remove it.

Reply:
I am going to remove it, as it is no longer needed.